lib/dispatch.go

```diff
@@ -97,25 +98,34 @@ func (inst *Instance) Dispatch(ctx context.Context, method string, param interfa
 		// TODO(dustmop): If the method wrote to our internal data structures, like
 		// refstore, logbook, etc, serialize and commit those changes here
 
-		// Validate the return values. This shouldn't fail as long as all method
-		// implementations are declared correctly
-		if len(outVals) != 2 {
-			return nil, fmt.Errorf("wrong number of return args: %d", len(outVals))
+		// Validate the return values.
+		if len(outVals) < 1 || len(outVals) > 3 {
+			return nil, nil, fmt.Errorf("wrong number of return values: %d", len(outVals))
 		}
-
 		// Extract the concrete typed values from the method return
 		var out interface{}
-		out = outVals[0].Interface()
-		errVal := outVals[1].Interface()
+		var cur Cursor
+		// There are either 1, 2, or 3 output values:
+		//   1: func() (err)
+		//   2: func() (out, err)
+		//   3: func() (out, cur, err)
```

Comment:
We should get this comment in a place where `MethodSet` implementers can find it. I think a big 'ol comment above the methodSet interface